content/reliability/docs/architecting-for-static-stability.md

```diff
@@ -0,0 +1,19 @@
+# Architecting for static stability
+
+When an availability zone outage occurs, the best way to maintain application availability is to be prepared. Rely on Kubernetes’ static stability, rather than automated evacuation strategies. Configure applications with [pod anti affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity) and [topology spread constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints) to spread replicas across failure domains, and test to verify that replicas can handle the traffic surge during an outage.
+
+Kubernetes nodes have a periodic heartbeat every 40 seconds. This informs the Kubernetes control plane that the node is healthy, able to accept new pods from the scheduler, and continue to execute existing pods bound to the node. There are a number of cases where the node heartbeat will stop, including network partitioning, hardware failure, or shutdown of the underlying machine. These events may be isolated to a single node, or correlated across many nodes in the cluster.
+
+For an isolated availability event, pods should be terminated and rescheduled elsewhere to restore the desired available replicas of its owning controller. Kubernetes achieves this through [taint based eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions). After 40 seconds without a heartbeat, the node lifecycle controller applies a NoExecute taint to the node. By default, pods tolerate this NoExecute taint for 300 seconds, before they are terminated by the pod lifecycle controller. The owning controller launches a new pod, which is then scheduled to a healthy node.
+
+This behavior works well for isolated availability events, but what happens during a correlated event, such as an availability zone or regional power failure? When 55% of the nodes in an availability zone fail to send a heartbeat, the node lifecycle controller determines that the cluster is in a state of partial disruption and halts taint based eviction. Once the number of available nodes returns below this threshold, taint based eviction is restored. If an entire availability zone goes down, why doesn’t Kubernetes to taint all of the nodes and migrate the workloads to another availability zone? Architecting for availability zone evacuation (e.g. running at a high utilization with the plan to launch new capacity during an AZ failure) contains multiple bad assumptions that risk application availability.
+
+First, is the assumption that capacity will be available during the outage. If the event is correlated, workloads across accounts may be impacted. If each of these workloads evacuate at the same time, this will result in a large spike in demand for capacity in the other availability zones. In many cases, this can be absorbed by large capacity pools in EC2, but risks increase for higher demand instance types and smaller regions.
```

Comment:
```suggestion
First, is the assumption that capacity will be available in other Availability Zones during the outage. If the event is correlated, other customer workloads may be impacted as well. If every workload evacuates at the same time, it will cause a large spike in demand for capacity in the other Availability Zones. In many cases, this can be absorbed by large capacity pools in EC2, but risks increase for higher demand instance types and smaller regions.
```